### PR TITLE
Add new view for AWS ECS health check extension.

### PR DESCRIPTION
**Pull request by skyduo**
**Description:**
This is a new view we are going to use in AWS ECS health check extension to collect the failures of exporters.

**Link to tracking Issue:** 
https://github.com/open-telemetry/opentelemetry-collector/issues/2573

**Testing:** 
`make otelcol` and run executable file

**Documentation:** 
https://docs.google.com/document/d/1SpUMsWA2DeaoVazeQ8uEc1Wvu5LphmQU_TjzLmuJ4QM/edit#heading=h.rs1luwizct2w

### CODE REVIEW

**Comment by bogdandrutu** [comment]
Please sign the CLA

**Review by hossain-rayhan — APPROVED** [review]

**Comment by hossain-rayhan** [comment]
Hi @bogdandrutu, this is required for our ECS Health Check. Can you please have a quick look. 

CC: @alolita

**Comment by skyduo** [comment]
Major changes in this pr also need review https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/4451

**Review by Aneurysm9 — APPROVED** [review]

File: internal/obsreportconfig/obsreportconfig.go [hunk 1]
```diff
@@ -88,6 +88,14 @@ func allViews() *ObsMetrics {
 	tagKeys = []tag.Key{obsmetrics.TagKeyExporter}
 	views = append(views, genViews(measures, tagKeys, view.Sum())...)
 
+	errorNumberView := &view.View{
+		Name:        "ExporterFailedNumber",
```
Comment (bogdandrutu) [inline internal/obsreportconfig/obsreportconfig.go:92]: Replace "ExporterFailedNumber" with "exporter/send_failed_requests"? To be consistent with other views.

**Comment by bogdandrutu** [comment]
Please rebase to fix incompatible prs check

File: internal/obsreportconfig/obsreportconfig.go (hunk 1 above)
Comment (skyduo) [reply internal/obsreportconfig/obsreportconfig.go:92]: good point, will change to that

File: internal/obsreportconfig/obsreportconfig.go [hunk 2]
```diff
@@ -88,6 +88,14 @@ func allViews() *ObsMetrics {
 	tagKeys = []tag.Key{obsmetrics.TagKeyExporter}
 	views = append(views, genViews(measures, tagKeys, view.Sum())...)
 
+	errorNumberView := &view.View{
+		Name:        "exporter/send_failed_requests",
```
Comment (bogdandrutu) [inline internal/obsreportconfig/obsreportconfig.go:92]: Use the prefix constant to ensure no typos

Comment (skyduo) [reply internal/obsreportconfig/obsreportconfig.go:92]: do you mean use `ExporterPrefix` like other views? actually we don't really care which exporter does the failure come from, so i think we can keep use this general view name.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**